easybuild/easyblocks/m/mxnet.py

```diff
@@ -0,0 +1,191 @@
+            regex_subs.append((r"(?P<var>MKLML_ROOT=).*$", r"# \g<var>%s" % os.environ["MKLROOT"]))
+        elif toolchain_blas in ['ACML', 'ATLAS']:
+            blas = "atlas"
+        elif toolchain_blas == 'OpenBLAS':
+            blas = "openblas"
+        elif toolchain_blas is None:
+            # This toolchain has no BLAS library
+            raise EasyBuildError("No BLAS library found in the toolchain")
+
+        regex_subs.append((r'USE_BLAS =.*', 'USE_BLAS = %s' % blas))
+
+        if get_software_root('NNPACK'):
+            regex_subs.append(("USE_NNPACK = 0", "USE_NNPACK = 1"))
+
+        apply_regex_substitutions('config.mk', regex_subs)
+
+        super(EB_MXNet, self).configure_step()
+
+    def install_step(self):
+        """Specify list of files to copy"""
+        self.cfg['files_to_copy'] = ['bin', 'include', 'lib',
+                                     (['dmlc-core/include/dmlc', 'nnvm/include/nnvm'], 'include')]
+        super(EB_MXNet, self).install_step()
+
+    def extensions_step(self):
+        """Build & Install both Python and R extension"""
+        # we start with the python bindings
+        self.py_ext = PythonPackage(self, {'name': 'mxnet'})
+        self.py_ext.module_generator = self.module_generator
+        self.py_ext.src = os.path.join(self.mxnet_src_dir, "python")
+        os.chdir(self.py_ext.src)
+
+        self.py_ext.prerun()
+        self.py_ext.run(unpack_src=False)
+        self.py_ext.postrun()
+
+        # next up, the R bindings
+        self.r_ext = RPackage(self, {'name': 'R-package'})
+        self.r_ext.module_generator = self.module_generator
+        self.r_ext.src = os.path.join(self.mxnet_src_dir, "R-package")
+        try:
+            os.chdir(self.r_ext.src)
+            mkdir("inst")
+            symlink(os.path.join(self.installdir, "lib"), "inst/libs")
+            symlink(os.path.join(self.installdir, "include"), "inst/include")
+        except IOError, err:
+            raise EasyBuildError("Failed to prepare the 'inst' directory for the R bindings: %s", err)
+
+        namespace = ["import(Rcpp)", "import(methods)", ""]
+        write_file("NAMESPACE", "\n".join(namespace))
+        self.r_ext.prerun()
+        self.r_ext.run()
+        self.r_ext.postrun()
+
+    def sanity_check_step(self):
+        """Check for main library files for MXNet"""
+        custom_paths = {
+            'files': ["lib/libmxnet.%s" % ext for ext in ['a', get_shared_lib_ext()]],
+            'dirs': [],
+        }
+        super(EB_MXNet, self).sanity_check_step(custom_paths=custom_paths)
+        self.py_ext.sanity_check_step()
+        self.r_ext.sanity_check_step()
+
+    def make_module_extra(self, *args, **kwargs):
+        """Custom variables for MXNet module."""
+        txt = super(EB_MXNet, self).make_module_extra(*args, **kwargs)
+        txt += self.py_ext.make_module_extra(*args, **kwargs)
+        txt += self.r_ext.make_module_extra(*args, **kwargs)
+        return txt
```

Comment:
The way you're using it is (clearly) not how it's intended to be used, since `make_module_extra` also defines `$EBROOT*` and `$EBVERSION*`.

You should just extend `txt` with update statements for `$PYTHONPATH` and `$R_LIBS` yourself, we don't have a better way of dealing with that unless we do some refactoring (e.g. flesh out the bit that defines `$PYTHONPATH` into a separate method that we can call, etc.)